test/built-ins/RegExp/prototype/Symbol.match/builtin-coerce-global.js

```diff
@@ -0,0 +1,97 @@
+// Copyright (C) 2015 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+description: Type coercion of `global` property value
+es6id: 21.2.5.6
+info: >
+    [...]
+    5. Let global be ToBoolean(Get(rx, "global")).
+    6. ReturnIfAbrupt(global).
+    7. If global is false, then
+       a. Return RegExpExec(rx, S).
+
+    21.2.5.2.1 Runtime Semantics: RegExpExec ( R, S )
+
+    [...]
+    7. Return RegExpBuiltinExec(R, S).
+
+    21.2.5.2.2 Runtime Semantics: RegExpBuiltinExec ( R, S )
+
+    [...]
+    6. Let global be ToBoolean(Get(R, "global")).
+features: [Symbol.match]
+---*/
+
+var r = /./;
+var val, result;
+Object.defineProperty(r, 'global', {
+  get: function() {
+    return val;
+  }
+});
+
+val = false;
+result = r[Symbol.match]('ab');
```

Comment:
My concern is readability. Since we have multiple assertions, it is harder here compared to other tests to match the expected outcome for a given input.

If we had for example two functions called `assertCoercible` and `assertNotCoercible`, it would be easy to scan through the file since it would say:

    function assertNotCoercible(global) {
        val = global;

        result = r[Symbol.match]('ab');

        assert.notSameValue(result, null);
        assert.sameValue(result.length, 1);
        assert.sameValue(result[0], 'a');
    }

    ...

    assertNotCoercible(false);
    assertNotCoercible('');
    ...
    assertCoercible(true);
    assertCoercible('truthy');

I'm fine with the version of the code now because, as you said, I don't expect people to read it often.